src/plugins/data/public/query/query_string/dataset_service/types.ts

```diff
@@ -83,5 +83,5 @@ export interface DatasetTypeConfig {
   /**
    * Returns a list of sample queries for this dataset type
    */
-  getSampleQueries?: (dataset: Dataset, language: string) => any;
+  getSampleQueries?: (dataset?: Dataset, language?: string) => Promise<any> | any;
```

Comment:
Have we updated all callers to handler promises with this interface change?

Pre-existing, but can we be more specific than any?

Reply:
Yes it's only being called in `no_results.tsx` and we modified it so that it handles the promise